torch/_meta_registrations.py

```diff
@@ -516,6 +516,76 @@ def avg_pool2d_backward_shape_check(
     check_dim_size(gradOutput, ndim, ndim - 1, outputWidth)
 
 
+# Don't override the C++ registration.
+@register_meta(aten.avg_pool2d_backward.default, register_dispatcher=False)
+def meta_avg_pool2d_backward(
+    gradOutput_,
+    input,
+    kernel_size,
+    stride,
+    padding,
+    ceil_mode,
+    count_include_pad,
+    divisor_override,
+):
+    # From aten/src/ATen/native/AveragePool2d.cpp structured kernel meta func.
+    check(
+        len(kernel_size) == 1 or len(kernel_size) == 2,
+        lambda: "avg_pool2d: kernel_size must either be a single int, or a tuple of two ints",
+    )
+    kH = kernel_size[0]
+    kW = kH if len(kernel_size) == 1 else kernel_size[1]
+    check(
+        len(stride) == 0 or len(stride) == 1 or len(stride) == 2,
+        lambda: "avg_pool2d: stride must either be omitted, a single int, or a tuple of two ints",
+    )
+    dH = kH if len(stride) == 0 else stride[0]
+    dW = kW if len(stride) == 0 else dH if len(stride) == 1 else stride[1]
+    check(
+        len(padding) == 1 or len(padding) == 2,
+        lambda: "avg_pool2d: padding must either be a single int, or a tuple of two ints",
+    )
+    padH = padding[0]
+    padW = padH if len(padding) == 1 else padding[1]
```

Comment:
I think this code is my fault, but just noting that we can do wayyyy better in Python now lol

Reply:
Since the structure here is to follow the c++ code, not sure if we want to change it or not.
Do you think that preserving the c++ structure is better or making these nice python functions?